src/compiler/scala/reflect/macros/contexts/Typers.scala

```diff
@@ -41,7 +41,14 @@ trait Typers {
 
   def inferImplicitValue(pt: Type, silent: Boolean = true, withMacrosDisabled: Boolean = false, pos: Position = enclosingPosition): Tree = {
     macroLogVerbose(s"inferring implicit value of type $pt, macros = ${!withMacrosDisabled}")
-    universe.analyzer.inferImplicit(universe.EmptyTree, pt, false, callsiteTyper.context, silent, withMacrosDisabled, pos, (pos, msg) => throw TypecheckException(pos, msg))
+    // If there is an open implicit in the context, use its tree as the one for which the implicit needs to be inserted.
+    // Otherwise, use the empty tree. This prevents false divergent implicit errors like the one in
+    // https://github.com/scala/bug/issues/10398.
+    val tree = callsiteTyper.context.openImplicits match {
+      case oi :: _ => oi.tree
+      case _ => universe.EmptyTree
+    }
+    universe.analyzer.inferImplicit(tree, pt, false, callsiteTyper.context, silent, withMacrosDisabled, pos, (pos, msg) => throw TypecheckException(pos, msg))
```

Comment:
Another possibility is to use `Ident(NoSymbol.newTermSymbol())` as the tree here. That would have the advantage of only requiring changes to the macro system, which lessens the risk of unexpected regressions.